parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh

```diff
@@ -10,6 +14,33 @@ removeContainerd() {
     apt_get_purge 10 5 300 moby-containerd
 }
 
+removeCurl() {
+    apt_get_purge 10 5 300 curl
+}
+
+installLatestCurlManuallyIfNotPresent() {
+    # Too Risky to manage within the components.json file since curl is required for the bootstrap logic to work
+    # currently this logic is only needed for 18.04, we will be able to remove this logic once we no longer support 18.04
+    
+    # renovate: datasource=github-releases depName=curl/curl
+    version="8.9.0"
+
+    if curl -V | grep $version > /dev/null; then
+        echo "curl is already installed and at the right version, skipping manual installation"
+        return
+    fi
+    
+    deb_file="/tmp/curl.deb"
+    removeCurl || exit $ERR_CURL_REMOVE_TIMEOUT
+    retrycmd_if_failure 10 5 10 wget https://curl.haxx.se/download/curl-${version}.tar.gz -O $deb_file || exit $ERR_CURL_DOWNLOAD_TIMEOUT
```

Comment:
does curl exist within microsoft's linux package repo? (packages.microsoft.com), or on some other microsoft-managed mirror?

if so we should use that rather than `curl.haxx.se` I think

Reply:
it might exists but the version wouldnt be good, the problem is that the latest available version for 1804 a a bug, so a more recent version compiled with the deps found in 1804, must be used.  I don't expect to be able to find such a package. but I can look